portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Policies/Policies.tsx

```diff
@@ -175,53 +177,83 @@ const Policies: React.FC = () => {
             const [apiPoliciesResponse, commonPoliciesResponse] = response;
-            const apiSpecificPolicies = apiPoliciesResponse.body.list;
-            const commonPolicies = commonPoliciesResponse.body.list;
-            const mergedList = [...commonPolicies, ...apiSpecificPolicies];
+            const apiSpecificPoliciesList = apiPoliciesResponse.body.list;
+            const commonPoliciesList = commonPoliciesResponse.body.list;
+            const mergedList = [...commonPoliciesList, ...apiSpecificPoliciesList];
 
             // Get all common policies and API specific policies
             setAllPolicies(mergedList);
 
-            let unionByPolicyDisplayName;
+            let apiPolicyByPolicyDisplayName;
+            let commonPolicyByPolicyDisplayName;
             if (showMultiVersionPolicies) {
-                // Get the union of policies depending on the policy display name and version
-                unionByPolicyDisplayName = [...mergedList
-                    .reduce((map, obj) => map.set(obj.name + obj.version, obj), new Map()).values()];
+                // Get the policies depending on the policy display name and version
+                apiPolicyByPolicyDisplayName = [...apiSpecificPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name + obj.version, obj), new Map()).values()];
+                commonPolicyByPolicyDisplayName = [...commonPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name + obj.version, obj), new Map()).values()];
             } else {
-                // Get the union of policies depending on the policy display name
-                unionByPolicyDisplayName = [...mergedList
-                    .reduce((map, obj) => map.set(obj.name, obj), new Map()).values()];
+                // Get the policies depending on the policy display name
+                apiPolicyByPolicyDisplayName = [...apiSpecificPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name, obj), new Map()).values()];
+                commonPolicyByPolicyDisplayName = [...commonPoliciesList
+                    .reduce((map: Map<string, Policy>, obj: Policy) => 
+                        map.set(obj.name, obj), new Map()).values()];
             }
-            unionByPolicyDisplayName.sort(
+            apiPolicyByPolicyDisplayName.sort(
                 (a: Policy, b: Policy) => a.name.localeCompare(b.name))
+            commonPolicyByPolicyDisplayName.sort(
+                (a: Policy, b: Policy) => a.name.localeCompare(b.name))
+
+            let filteredApiPolicyByGatewayTypeList = null;
+            let filteredCommonPolicyByGatewayTypeList = null;
             
-            let filteredByGatewayTypeList = null;
             if (!isChoreoConnectEnabled) {
```

Comment:
This can be simplified as,

```js
let gatewayType = isChoreoConnectEnabled ? 'ChoreoConnect' : 'Synapse';
// Get relevant gateway supported policies
filteredApiPolicyByGatewayTypeList = apiPolicyByPolicyDisplayName.filter(
    (policy: Policy) => policy.supportedGateways.includes(gatewayType));
filteredCommonPolicyByGatewayTypeList = commonPolicyByPolicyDisplayName.filter(
    (policy: Policy) => policy.supportedGateways.includes(gatewayType));
```